src/Annotation/EloquaAppCloudInteractiveResponder.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Drupal\eloqua_app_cloud\Annotation;
+
+use Drupal\Component\Annotation\Plugin;
+
+/**
+ * Defines a Eloqua AppCloud Interactive Responder
+ * as a base for other plugin types.
+ *
+ * @see \Drupal\eloqua_app_cloud\Plugin\EloquaAppCloudInteractiveResponderBase
+ * @see plugin_api
+ *
+ * @Annotation
+ */
+class EloquaAppCloudInteractiveResponder extends Plugin {
+
+  /**
+   * The plugin ID.
+   *
+   * @var string
+   */
+  public $id;
+
+  /**
+   * The label of the plugin.
+   *
+   * @var \Drupal\Core\Annotation\Translation
+   *
+   * @ingroup plugin_translatable
+   */
+  public $label;
+
+  /**
+   * The list of fields required by the plugin.
+   *
+   * These must be presented in the form:
+   * fieldList = {
+   *   "EmailAddress" = "{{Contact.Field(C_EmailAddress)}}"
+   *  },
+   *
+   * The key (i.e. EmailAddress), is arbitrary, but must be used consistently and is case sensitive.
+   * The value (i.e. Contact.Field(C_EmailAddress)), is the internal Eloqua field description.
+   * Extracting this value from Eloqua
+   * (possibly by using a call to https://secure.p01.eloqua.com/API/bulk/2.0/contacts/fields )
+   * is left as an exercise for the reader.
+   *
+   * @var array
+   *
+   */
+  public $fieldList;
+
+  /**
+   * The API type (contacts or customObject).
+   *
+   * @var string
+   */
+  public $api;
+
+
+  /**
+   * The name of the queue worker this plugin requires.
+   *
+   * @var string
+   */
+  public $queueWorker;
```

Comment:
Same comment from previously.  Seems like there's a 1:1 relationship between queueWorker and Step/Content plugin.  Do you need to require this as a property on the annotation, or can you get away with one of:

- Keeping the `queueWorker()` method, but hard-code the queueworker name in each base plugin (`EloquaAppCloudDecisionResponderBase`, `EloquaAppCloudActionResponderBase`, `EloquaAppCloudContentResponderBase`).
- Or, keep it more or less as-is, but fall-back to a hard-coded value (like above). But allow the plugin to define its own custom queueworker in the annotation (instead of in a method).

Reply:
I am a bit confused as to the difference between your suggestions. 
The plugin->queueWorker() method is currently just a convenience to pull the value from the plugin's annotation. I do want to maintain the ability for implementations to define custom queueWorkers, so... 

Are you suggesting that I provide a hard coded fallback in the base class that an implementation's annotation value would override if provided?

Reply:
I think my main rub is that you can override via annotation OR via a method with no clear reason to the end-user which is preferable.  Also that you could possibly do both with unintuitive results.  Too much choice!

I think I'd recommend simplifying the annotation and keeping the logic in methods.  Still allows overrides via implementor and makes the annotation lighter.